composer/datasets/webdataset_utils.py

```diff
@@ -15,11 +15,11 @@
 from tqdm import tqdm
 
 if TYPE_CHECKING:
-    from webdataset import WebDataset
+    from webdataset import WebDataset  # type: ignore
```

Comment:
this shouldn't throw a pyright error, curious the reason for this? Prefer not to litter the codebase with type ignores..

Reply:
Same thing as here: https://github.com/mosaicml/composer/pull/1001/files#r866082405

Reply:
Removed the `# type: ignore`; switched the pyproject.toml to ignoring the error